src/Analyser/MutatingScope.php

```diff
@@ -2772,7 +2774,7 @@ private function enterFunctionLike(
 				}
 			}
 			$parameterNode = new Variable($parameter->getName());
-			$expressionTypes[$paramExprString] = ExpressionTypeHolder::createYes($parameterNode, $parameterType);
+			$expressionTypes[$paramExprString] = ExpressionTypeHolder::createYes($parameterNode, $this->ruleLevelHelper->transformCommonType($parameterType));
```

Comment:
I don't like this because Scope shouldn't depend on rules. Type inference is rules and rule level-agnostic. Rules already depend on Scope. There shouldn't be a dependency in the other direction.